### ISSUE TITLE
[Flaky Test]: TestEndpointSecurityNonDefaultBasePath, TestEndpointSecurityUnprivileged – version conflict, document already exists

### ISSUE THREAD

**Issue by rdner** [elastic/elastic-agent#4883]
### Failing test case

TestEndpointSecurityNonDefaultBasePath, TestEndpointSecurityUnprivileged

### Error message

version conflict, document already exists

### Build

https://buildkite.com/elastic/elastic-agent-extended-testing/builds/441

### OS

Linux

### Stacktrace and notes

```markdown
endpoint_security_test.go:614: Installing Elastic Defend
    endpoint_security_test.go:615: Templating endpoint package policy request
    endpoint_security_test.go:615: POST /api/fleet/package_policies
    endpoint_security_test.go:615: Error installing fleet package: Error installing endpoint 8.11.0: version_conflict_engine_exception
        	Root causes:
        		version_conflict_engine_exception: [data_frame_transform_state_and_stats-endpoint.metadata_current-default-8.11.0]: version conflict, document already exists (current version [1])
    endpoint_security_test.go:616: 
        	Error Trace:	/home/ubuntu/agent/testing/integration/endpoint_security_test.go:616
        	Error:      	Received unexpected error:
        	            	error installing fleet package: Error installing endpoint 8.11.0: version_conflict_engine_exception
        	            		Root causes:
        	            			version_conflict_engine_exception: [data_frame_transform_state_and_stats-endpoint.metadata_current-default-8.11.0]: version conflict, document already exists (current version [1])
        	Test:       	TestEndpointSecurityUnprivileged
        	Messages:   	Policy Response was: {{ 0 false [] { }     }}
```

**Comment by rdner** [comment]
Possibly related to:

* https://github.com/elastic/elastic-agent/issues/4158
* https://github.com/elastic/elastic-agent/issues/3524

**Comment by rdner** [comment]
@elastic/fleet could you have a look?

**Comment by rdner** [comment]
Another failure in https://buildkite.com/elastic/elastic-agent-extended-testing/builds/727#01902afb-0d73-4154-b29f-1353b660d38c

**Comment by criamico** [comment]
It doesn't look related to any of the changes I've worked on recently. It seems another instance of the errors in https://github.com/elastic/elastic-agent/issues/4158 that for some reason are happening again even after the [fix](https://github.com/elastic/kibana/pull/177380). [This comment](https://github.com/elastic/elastic-agent/issues/4158#issuecomment-1954478336) seems to explain very well what the error is on transforms side. We can probably open another ticket to take a better look at it, it's strange that the error is reoccurring even though it was [explicitly ignored](https://github.com/elastic/kibana/blob/c34bb46c8aeb59dca0c92132dd34d2d9fca68728/x-pack/plugins/fleet/server/services/epm/elasticsearch/transform/install.ts#L742-L750).

**Comment by blakerouse** [comment]
@criamico This error seems to be specific to Fleet, but not something we have seen in over a month. Do you think something has changed that caused this error to go away?

I think it might be best to close this if its not reproducible.

**Comment by jlind23** [comment]
Agree with @blakerouse, closing this for now as it seems it didn't happen for the past months.

**Comment by pkoutsovasilis** [comment]
another occurance [here](https://buildkite.com/elastic/elastic-agent-extended-testing-bk/builds/2036#019528ca-e402-4b11-8138-4f0269bb5560/342-1008) re-opening

**Comment by pkoutsovasilis** [comment]
we have another test, namely `TestEndpointAgentServiceMonitoring` failing, but I believe the reason is the same [here](https://buildkite.com/elastic/elastic-agent-extended-testing-bk/builds/2036#019528ca-e401-47c1-a319-af962b46fcc4/342-986)

**Comment by swiatekm** [comment]
I got:

```
error installing fleet package: Unable to remove package with existing package policy(s) in use by agent(s)
```

in here: https://buildkite.com/elastic/elastic-agent/builds/17907#01955c4b-fdab-4c73-a79b-a26f81953205.

Not the same error, but looks related.

**Comment by cmacknz** [comment]
Another one https://buildkite.com/elastic/elastic-agent/builds/18059#0195629b-e7ef-4ceb-be3b-057758e26fa6 from https://github.com/elastic/elastic-agent/pull/7179

**Comment by cmacknz** [comment]
https://github.com/elastic/elastic-agent/issues/7221 is a different error but might be the same root cause

**Comment by criamico** [comment]
I created a new ticket in Kibana to track this issue: https://github.com/elastic/kibana/issues/213337

**Comment by pkoutsovasilis** [comment]
@pazone @dliappis after looking this one again, this Test runs in all OSes and all architecture, and thus flakiness can be introduced by having this running in parallel as already mentioned during our team meeting (another occurrence of it [here](https://buildkite.com/elastic/elastic-agent/builds/18420#01957f5a-b947-46dd-9db5-0833b5979839/297-551))

**Comment by pazone** [comment]
Added the issue to the scope. Will draft a PR to verify the assumption. Ideally, we better eliminate the probability of a bug.

**Comment by v1v** [comment]
Hi @pazone, I just discovered it, and Blake is assigned to it.  are we planning to work on this task in our current iteration? If so, perhaps we should create a new GitHub issue to outline what needs to be done, so that whoever takes on the task has a clear understanding of the work involved. What do you think?

**Comment by pazone** [comment]
@pkoutsovasilis [The issue](https://github.com/elastic/ingest-dev/issues/5242) is created with high priority. The next iteration starts in 2 days. The implementation will start this week

**Comment by pkoutsovasilis** [comment]
> Multiple ESS stacks will probably eliminate the flakiness but significantly increase the testing cost. And it's also important to figure out the actual reason for the flakiness.

I hear you @pazone , if cost a concern could you please provide some figures about the impact of doing multiple ESS stacks. Then we can run these with @ebeahan, and until the product team looks into the current limitations of using a single ESS stack, we may or may not resort to using multiple ESS stacks.

**Comment by blakerouse** [comment]
Could we just have the integration framework install the endpoint package before running any tests? That would ensure that it is only installed once into Kibana and then the tests could run.

**Comment by pazone** [comment]
@blakerouse Sure. We have a dedicated step that runs the stack. We can install the package within this step when the stack is ready.

**Comment by pkoutsovasilis** [comment]
closing this in favour of https://github.com/elastic/elastic-agent/issues/7529
